include/alpaka/core/Assert.hpp

```diff
@@ -20,8 +20,15 @@
 
 #ifdef ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST
 #    define ALPAKA_ASSERT_OFFLOAD(EXPRESSION) ALPAKA_ASSERT(EXPRESSION)
+#elif defined __AMDGCN__ && (!defined NDEBUG)
+#    define ALPAKA_ASSERT_OFFLOAD(EXPRESSION)                                                                         \
+        if(!(EXPRESSION))                                                                                             \
+            __builtin_trap();                                                                                         \
+        do                                                                                                            \
+        {                                                                                                             \
+        } while(false)
```

Comment:
please move line`25,26` into the `do` body. This is a common pattern for marcos and make the macro save to be used within `if` condition branches.

Reply:
```C++
#define ALPAKA_ASSERT_OFFLOAD(EXPRESSION) \
do \
{ \
if(!(EXPRESSION)) \
  __builtin_trap(); \
} \
while(false)
```
```

Reply:
Ahh, Ok. Done.